src/chainstate/burn/operations/transfer_stx.rs

```diff
@@ -0,0 +1,324 @@
+struct ParsedData {
+    transfered_ustx: u128,
+    memo: Vec<u8>,
+}
+
+impl TransferStxOp {
+    #[cfg(test)]
+    pub fn new(
+        sender: &StacksAddress,
+        recipient: &StacksAddress,
+        transfered_ustx: u128,
+    ) -> TransferStxOp {
+        TransferStxOp {
+            sender: sender.clone(),
+            recipient: recipient.clone(),
+            transfered_ustx,
+            memo: vec![],
+            // to be filled in
+            txid: Txid([0u8; 32]),
+            vtxindex: 0,
+            block_height: 0,
+            burn_header_hash: BurnchainHeaderHash([0u8; 32]),
+        }
+    }
+
+    fn parse_data(data: &Vec<u8>) -> Option<ParsedData> {
+        /*
+            Wire format:
+            0      2  3                             19        80
+            |------|--|-----------------------------|---------|
+             magic  op     uSTX to transfer (u128)     memo (up to 61 bytes)
+
+             Note that `data` is missing the first 3 bytes -- the magic and op have been stripped
+
+             The values ustx to transfer are in big-endian order.
+        */
+
+        if data.len() < 16 {
+            // too short
+            warn!(
+                "TransferStxOp payload is malformed ({} bytes, expected >= {})",
+                data.len(),
+                16
+            );
+            return None;
+        }
+
+        if data.len() > (61 + 16) {
+            // too long
+            warn!(
+                "TransferStxOp payload is malformed ({} bytes, expected <= {})",
+                data.len(),
+                16 + 61
+            );
+            return None;
+        }
+
+        let transfered_ustx = parse_u128_from_be(&data[0..16]).unwrap();
+        let memo = Vec::from(&data[16..]);
+
+        Some(ParsedData {
+            transfered_ustx,
+            memo,
+        })
+    }
+
+    pub fn get_sender_txid(tx: &BurnchainTransaction) -> Result<&Txid, op_error> {
+        match tx.get_input_tx_ref(0) {
+            Some((ref txid, vout)) => {
+                if *vout != 1 {
+                    warn!(
+                        "Invalid tx: TransferStxOp must spend the second output of the PreStacksOp"
+                    );
+                    Err(op_error::InvalidInput)
+                } else {
+                    Ok(txid)
+                }
+            }
+            None => {
+                warn!("Invalid tx: TransferStxOp must have at least one input");
+                Err(op_error::InvalidInput)
+            }
+        }
+    }
+
+    pub fn from_tx(
+        block_header: &BurnchainBlockHeader,
+        tx: &BurnchainTransaction,
+        sender: &StacksAddress,
+    ) -> Result<TransferStxOp, op_error> {
+        TransferStxOp::parse_from_tx(
+            block_header.block_height,
+            &block_header.block_hash,
+            tx,
+            sender,
+        )
+    }
+
+    /// parse a StackStxOp
+    /// `pox_sunset_ht` is the height at which PoX *disables*
```

Comment:
Copypasta comment? ;)

Reply:
Addressed in https://github.com/blockstack/stacks-blockchain/pull/2129/commits/b4c7b8878ad442a81a5fe47b6e94de287de49305